File: zeitara.domain/Entities/Purchase.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Purchase
    {
        [Key]
        public int PurchaseId { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public int ProductId { get; set; }

        [Required]
        [Range(1, int.MaxValue)]
        public int Quantity { get; set; }

        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "Completed";

        [Required]
        public DateTime PurchaseDate { get; set; }
    }
}

File: zeitara_webapp/src/app/shared/models/purchase/purchase.ts
// shared/models/purchases/purchase.model.ts
import { Product } from '../products/products'; // Import your existing Product interface

// Request DTO (matches backend PurchaseRequestDto)
export interface PurchaseRequest {
  userId: number;
  productId: number;
  quantity: number;
}

// Response DTO (matches backend PurchaseResponseDto)
export interface PurchaseResponse {
  purchaseId: number;
  productId: number;
  quantity: number;
  status: string;
  purchaseDate: Date;
  productDisplayName: string;
  imageUrl: string;
  isLiked: boolean;
  isInCart: boolean;
  isPurchased: boolean;
  userRating?: number;
}

// For the cart purchase response (matches InteractionResponseDto)
export interface InteractionResponse {
  message: string;
}
